src/components/Input/Input.tsx

```diff
@@ -1,112 +1,138 @@
+import { useMutation } from '@tanstack/react-query';
+import Container from '../utility-components/Container';
+import { ResponseType } from '~/types';
+import { fetchSummaryData } from '~/query/fetch-summary-data';
+import Image from 'next/image';
+import loaderGif from '../../assets/loader.gif';
+import errorIcon from '../../assets/error.png';
 export type handleFormSubmitType = (
-  event: React.SyntheticEvent,
-  type: ContentType,
-  wordLimit: number,
-  inputUrl?: string,
-  text?: string
+	event: React.SyntheticEvent,
+	type: ContentType,
+	inputUrl?: string,
+	text?: string
 ) => Promise<void>;
 
-const Input = () => {
-  const [inputTypeSelected, setInputTypeSelected] = useState<InputType>(
-    InputType.WEBSITE
-  );
-  const [summaryLength, setSummaryLength] = useState("200");
+const Input = ({
+	handleSummarize,
+}: {
+	handleSummarize: (newResult: ResponseType) => void;
+}) => {
+	const [inputTypeSelected, setInputTypeSelected] = useState<InputType>(
+		InputType.WEBSITE
+	);
+	const [summaryLength, setSummaryLength] = useState('200');
+
+	const { isLoading, mutate, error, isError } = useMutation({
+		mutationFn: fetchSummaryData,
+		onSuccess: (res) => {
+			console.log(res);
+			handleSummarize(res.openAiResponse);
+		},
+	});
+
+	const handleFormSubmit: handleFormSubmitType = async (
+		event,
+		type,
+		inputUrl,
+		text
+	) => {
+		event.preventDefault();
+		mutate({
+			url: inputUrl ?? '',
+			wordLimit: parseInt(summaryLength),
+			type,
+			text,
+		});
+	};
 
-  const handleFormSubmit: handleFormSubmitType = async (
-    event,
-    type,
-    wordLimit,
-    inputUrl,
-    text
-  ) => {
-    event.preventDefault();
-    try {
-      const request: RequestBody = {
-        url: inputUrl ?? "",
-        wordLimit,
-        type,
-        text,
-      };
-      const res = await fetchSummaryData(request);
+	// TODO: Refactor loading and error states
+	if (isLoading)
+		return (
+			<Container>
+				<div className='flex min-h-[30rem] items-center justify-center pb-10 text-center'>
+					<div>
+						<Image src={loaderGif} alt='my gif' height={200} width={200} />
+					</div>
+				</div>
+			</Container>
+		);
 
-      const existingData = localStorage.getItem("summaries");
-      if (existingData) {
-        const existingDataArr: { [key: string]: string }[] =
-          JSON.parse(existingData);
-        existingDataArr.push(res);
-        localStorage.setItem("summaries", JSON.stringify(existingDataArr));
-      } else {
-        localStorage.setItem("summaries", JSON.stringify([res]));
-      }
-    } catch (error) {
-      console.error(error);
-    }
-  };
+	if (isError)
+		return (
+			<Container>
+				<div className='flex min-h-[30rem] flex-col items-center justify-center text-center'>
+					<div>
+						<Image src={errorIcon} alt='error' height={200} width={200} />
+					</div>
+					<h2
+						onClick={() => window.location.reload()}
+						className='top-10 cursor-pointer text-heading3 font-bold'>
```

Comment:
instead of a reload here we might just want to retry the same request or show that there was an error and have a button to go back to previous state?

Reply:
oh yes, i just added this just for the preview haha.. i think there is tons of refactoring to do and i'll fix this when i push the changes